Replace pyvista 3d plot with matplotlib animation in Precip example

LSchueler opened the pull request:
I've updated the precipitation example to now show the time evolution of the 2D P-field as an animated figure.

MuellerSeb reviewed (changes requested):
This animation look amazing! Thanks for adding!

I had a second though on spatio-temporal modeling:
We should create a separate example folder/category for "Spatio-Temporal Modelling".

Precipitation is one of the prime application examples, so your work could shine here :wink:
But I'd split your example in two, one 1D and the other one 2D.

In order to make the spatio-temporal people happy, I'd suggest to come up with some conventions:

1. What we provide at the moment is a so called *"metric"* spatio-temporal model. We should mention that.
2. time dimension should be the last one:
    - common in the spatio-temporal literature
    - Since `anis` in the `CovModel` will be filled up from the right, a passed `st_anis` for the space-time anisotropy of a metric model will be used correctly: (lucky we made a good decision here)
        ```python
        st_dim = 2 + 1  # spatial dimension of two and one time dimension
        st_anis = 0.4  # space-time anisotropy given in d / km
        # an exponential variogram with a corr. lengths of 5km, 5km and 2d
        model = gs.Exponential(dim=st_dim, var=1.0, len_scale=5.0, anis=st_anis)
        ```
    - since we now provide arbitrary dimensions with the special definition of rotation angles, the first rotation angles for the first n dimension in a (n+1) dimension model coincide with the rotation angles of a n-dimensional model (lucky we made a good decision here as well) .. this means 3D spatio-temporal models work out of the box :wink:
3. We should again use the `pos` tuple/list like in the other examples. We can use another list for the time extension to make it more visible what happens:
    ```python
    pos, time = [x, y], [t]
    srf = gs.SRF(model, seed=seed)
    srf.structured(pos + time)
    ```
    PS: The `seed` should be passed to `SRF` directly, so we the modes are not sampled twice in this example.
    We can use this `pos` later on `ax.pcolormesh(*pos, srf.field ...`
    
What do you think about that?

LSchueler commented on examples/00_misc/03_precipitation.py:
```diff
@@ -149,9 +150,17 @@
+    return quad,
+
+fig, ax = plt.subplots()
+quad = ax.pcolormesh(x, y, srf.field[0,:,:].T, cmap="Blues", shading="gouraud")
```
I never really liked `imshow`. I don't know why, but consistency is a good argument.

LSchueler commented on examples/00_misc/03_precipitation.py:
```diff
@@ -149,9 +150,17 @@
+fig, ax = plt.subplots()
+quad = ax.pcolormesh(x, y, srf.field[0,:,:].T, cmap="Blues", shading="gouraud")
+cbar = fig.colorbar(quad)
+cbar.ax.set_ylabel(r"Precipitation $P$ / mm")
+ax.set_xlabel(r"$x$ / km")
+ax.set_ylabel(r"$y$ / km")
```
YEAH!!!

LSchueler commented on examples/00_misc/03_precipitation.py:
```diff
@@ -149,9 +150,17 @@
 
-mesh = srf.to_pyvista()
-mesh.threshold_percent(0.25).plot()
+def _update_ani(idx):
+    quad.set_array(srf.field[idx, :, :].T.ravel())
+    return quad,
```
Well, to give users a hint that it's not an `imshow` object ;-)

LSchueler commented:
Thanks for your input! Having had more to do with special relativity than with spatio-temporal geostatistics I'm not used to the notion of putting the time dimension at the end of the space-time-vector.

I've split and updated the example accordingly, but I didn't have time to include any introductory text yet.

On my local machine, the matplotlib animation is only shown as a static image in the sphinx-gallery. Any idea?

MuellerSeb commented on examples/09_spatio_temporal/02_precip_2d.py:
```diff
@@ -0,0 +1,74 @@
+
+# the dry periods
+threshold = 0.4
+srf.field[srf.field <= threshold] = 0.0
+
+# account for the skewness
+gs.transform.boxcox(srf, lmbda=0.5, shift=-1.0)
```
I think you can drop the threshold line. These `0.0` aren't zero after the boxcox transformation and this creates strange ouput in terms of precipitation (same problem in the 1D example).

The boxcox transformation is implemented to cut off values, that can't be transformed, so you should result in "realistic" values when solely applying the boxcox trafo.

MuellerSeb replied:
Or you do the following:
```python
lmbda = 0.5
gs.transform.boxcox(srf, lmbda=lmbda, shift=-1 / lmbda)
```

MuellerSeb replied:
Or we also implement the yeo-johnson transformation in the `transform` submodule

LSchueler replied:
Good catch, thanks!

MuellerSeb replied:
When #124 is merged, we can also rewrite that to use the box-cox normalizer.

MuellerSeb reviewed (changes requested):
Like it much more! :wink:
Only thing is the threshold setting, which doesn't work well with the boxcox transformation afterwards.

And a `README.rst` is missing in the new examples folder. The following should be sufficient:
```rst
Spatio-Temporal Modelling
=========================

Examples
--------
```

With this, the documentation compiles correctly and the animation is working.

LSchueler commented:
> Like it much more! wink
> Only thing is the threshold setting, which doesn't work well with the boxcox transformation afterwards.
> 
> And a `README.rst` is missing in the new examples folder. The following should be sufficient:
> 
> ```
> Spatio-Temporal Modelling
> =========================
> 
> Examples
> --------
> ```
> 
> With this, the documentation compiles correctly and the animation is working.

Yeah... I forgot to add the readme to the commit stage. But that means that something is not working with sphinx gallery on my local machine. Whatever, as long as it runs on RTD.

LSchueler commented:
I've updated the examples accordingly.

MuellerSeb reviewed (approved):
Yeah!
